conda_build/build.py

```diff
@@ -2584,8 +2584,11 @@ def construct_metadata_for_test(recipedir_or_package, config):
 
 
 def write_build_scripts(m, script, build_file):
-    with utils.path_prepended(m.config.host_prefix):
-        with utils.path_prepended(m.config.build_prefix):
+    # TODO: Prepending the prefixes here should probably be guarded by
+    #         if not m.activate_build_script:
+    #       Leaving it as is, for now, since we need a quick, non-disruptive patch release.
+    with utils.path_prepended(m.config.host_prefix, False):
+        with utils.path_prepended(m.config.build_prefix, False):
```

Comment:
Before removing this in a future release, we should add more test cases.
The test added in this PR only checks for the case when there is `build`+`host`, `activate=True` and no `outputs`.
Other configurations that should be tested:
- `activate=False`
- `build` and `host` merged
  - with `activate=True` 
  - with `activate=False` 
- recipes with `outputs`
  - and those for potential combinations of the aforementioned other cases.

I consider this out of scope for this PR since we should put out a patch release quickly and because those other cases touch other issues like https://github.com/conda/conda-build/issues/3993 -- that's another can of worms (one we should take care of, but not hastily in a small patch release).